src/main/java/edu/texas/social_computing/hospitals/Resident.java

```diff
@@ -33,4 +42,24 @@ public Resident getPartner() {
         return null;
     }
 
+    public boolean hasPartner() {
+        return this.partner != null;
+    }
+
+    /**
+     *
+     * @return the Resident that is matched this a Hospital that is lower on their preference list than their partner.
+     * The tie always goes to this
+     */
+    public Resident worsePlacedPartner() {
+        Hospital hospital = this.assignedHospital;
+        Hospital partnerHospital = this.partner.getAssignedHospital();
+        if(hospital == null && partnerHospital == null) return this;
+        if(hospital == null && partnerHospital != null) return this.partner;
+        if(hospital != null && partnerHospital == null) return this;
+        int preference = this.preferences.indexOf(hospital);
+        int partnerPreference = this.partner.getPreferences().indexOf(partnerHospital);
+        return preference >= partnerPreference ? this : this.partner;
```

Comment:
We might need to be careful with `>=` to make sure that it doesn't cause both partners to try and relocate if they get their same preference in different locations...

Reply:
Oh I see you took care of that case above :)